src/ucs/debug/log.c

```diff
@@ -164,30 +172,41 @@ static void ucs_log_file_rotate()
 
     for (idx = ucs_log_file_last_idx - 1; idx >= 0; --idx) {
         ucs_log_get_file_name(old_log_file_name,
-                              sizeof(old_log_file_name), idx);
+                              PATH_MAX, idx);
         ucs_log_get_file_name(new_log_file_name,
-                              sizeof(new_log_file_name), idx + 1);
+                              PATH_MAX, idx + 1);
 
         if (access(old_log_file_name, W_OK) != 0) {
-            ucs_fatal("unable to write to %s", old_log_file_name);
+            ucs_error("unable to write to %s", old_log_file_name);
+            goto err_out;
         }
 
         /* coverity[toctou] */
         ret = rename(old_log_file_name, new_log_file_name);
         if (ret) {
-            ucs_fatal("failed to rename %s to %s: %m",
+            ucs_error("failed to rename %s to %s: %m",
                       old_log_file_name, new_log_file_name);
+            goto err_out;
         }
 
-
         if (access(old_log_file_name, F_OK) != -1) {
-            ucs_fatal("%s must not exist on the filesystem", old_log_file_name);
+            ucs_error("%s must not exist on the filesystem", old_log_file_name);
+            goto err_out;
         }
 
         if (access(new_log_file_name, W_OK) != 0) {
-            ucs_fatal("unable to write to %s", new_log_file_name);
+            ucs_error("unable to write to %s", new_log_file_name);
+            goto err_out;
         }
     }
+
+    ucs_free(old_log_file_name);
+    ucs_free(new_log_file_name);
+
+err_out:
+    ucs_free(old_log_file_name);
```

Comment:
no need to free i guess, since there would be so many leaked resources aynways..